lib.sh

```diff
@@ -385,6 +422,7 @@ calculate_max_children() {
 }
 
 test_connection() {
+local non_fatal="$1"
```

Comment:
exactly what does this do?

Reply:
not needed now when we don't use test_connection in the install script anymore (the thing you added)

Reply:
I am not sure that test_connection is unused. In fact I did not add it as it was already there is both places. There is a test_connection.sh but also a test_connection() function in lib.sh. The fact that the test_connection change you made in nextcloud_install_production.sh to add "$@" as a parameter would seem to indicate that it is the internal lib.sh function used rather than the external script?

When I tested static_ip.sh just now, it definitely used the lib.sh function. for the static_ip.sh case, I need that test_connection function to return 1 or error and not exit 1 on error. The reason is that I need to loop if the test fails. If I don't loop, then the exit will skip out of the static_ip.sh back into nextcloud-startup-script.sh which is not what we want any longer as we want users to try again with working details (if they can).

So I think that lib.sh and static_ip.sh are correct now but not nextcloud_install_production.sh.